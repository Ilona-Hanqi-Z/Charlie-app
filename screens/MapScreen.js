/* This file contains the map assignment page. */

import React from 'react';
import { View, StyleSheet, Alert } from 'react-native';
import MapView from 'react-native-maps';
import { Marker } from 'react-native-maps';
import * as Permissions from 'expo-permissions';
import * as Location from 'expo-location';
import { LocationAccuracy } from 'expo-location';
import { Assignments } from '../model/assignment';

class MapScreen extends React.Component {
    constructor(props){
      super(props);

      this.state = {
        region:null,
      }

      this._getLocationAsync();
    }

    _getLocationAsync = async() => {
      let { status } = await Permissions.askAsync(Permissions.LOCATION);
      if(status !== 'granted'){
        console.log('Permission to access location was denied.')
      }

      let location = await Location.getCurrentPositionAsync({accuracy:LocationAccuracy.Highest});
      let region = {
        latitude: location.coords.latitude,
        longitude: location.coords.longitude,
        latitudeDelta: 0.045,
        longitudeDelta: 0.045
      }; 

      this.setState({region: region});
    }

    render(){
      let markers = [];
      for(let i = 0; i < Assignments.length; i++){
        markers.push(
          <Marker
            key = {i}
            coordinate={{
              latitude:Assignments[i].latitude,
              longitude:Assignments[i].longitude,
            }}
            icon={require('../assets/map_marker.png')}
            title={Assignments[i].title}
            description={Assignments[i].caption}
          />
        )
      }
      return (  
      <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
        
        <MapView
          initialRegion={this.state.region}
          showsUserLocation={true}
          showsCompass={true}
          rotateEnabled={false}
          style={styles.map}
        >
          { markers }

        </MapView>
      </View>
    );
   }
  }

export default MapScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
  map: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
  },
});
