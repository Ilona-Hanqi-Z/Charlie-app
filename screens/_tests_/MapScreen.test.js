import React from 'react';
import { render } from 'react-native-testing-library';
import MapScreen from '../MapScreen';

test('it renders all inputs as expected', ()=> {
    const { toJSON } = render(<MapScreen />);

    expect(toJSON()).toMatchSnapshot();
})