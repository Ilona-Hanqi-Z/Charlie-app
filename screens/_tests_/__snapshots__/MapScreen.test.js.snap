// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders all inputs as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <AIRMap
    handlePanDrag={false}
    initialRegion={null}
    onChange={[Function]}
    onLayout={[Function]}
    onMapReady={[Function]}
    onMarkerPress={[Function]}
    region={null}
    rotateEnabled={false}
    showsCompass={true}
    showsUserLocation={true}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <AIRMapMarker
      coordinate={
        Object {
          "latitude": undefined,
          "longitude": undefined,
        }
      }
      description="caption1"
      icon="file://assets/full/path/to/directory/name.type"
      onPress={[Function]}
      stopPropagation={false}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "position": "absolute",
          },
          undefined,
        ]
      }
      title="test 1"
    >
      <AIRMapCallout
        alphaHitTest={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "position": "absolute",
            },
            undefined,
          ]
        }
        tooltip={true}
      >
        <View>
          <View
            style={
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#fff",
                "borderColor": "#ccc",
                "borderRadius": 6,
                "borderWidth": 0.5,
                "flexDirection": "column",
                "padding": 15,
                "width": 150,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
            >
              1
              . 
              test 1
            </Text>
            <Text>
              caption1
            </Text>
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "borderTopColor": "#007a87",
                "borderWidth": 16,
                "marginTop": -0.5,
              }
            }
          />
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "borderTopColor": "#fff",
                "borderWidth": 16,
                "marginTop": -32,
              }
            }
          />
        </View>
      </AIRMapCallout>
    </AIRMapMarker>
    <AIRMapMarker
      coordinate={
        Object {
          "latitude": undefined,
          "longitude": undefined,
        }
      }
      description="caption2"
      icon="file://assets/full/path/to/directory/name.type"
      onPress={[Function]}
      stopPropagation={false}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "position": "absolute",
          },
          undefined,
        ]
      }
      title="test 2"
    >
      <AIRMapCallout
        alphaHitTest={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "position": "absolute",
            },
            undefined,
          ]
        }
        tooltip={true}
      >
        <View>
          <View
            style={
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#fff",
                "borderColor": "#ccc",
                "borderRadius": 6,
                "borderWidth": 0.5,
                "flexDirection": "column",
                "padding": 15,
                "width": 150,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
            >
              2
              . 
              test 2
            </Text>
            <Text>
              caption2
            </Text>
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "borderTopColor": "#007a87",
                "borderWidth": 16,
                "marginTop": -0.5,
              }
            }
          />
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "borderTopColor": "#fff",
                "borderWidth": 16,
                "marginTop": -32,
              }
            }
          />
        </View>
      </AIRMapCallout>
    </AIRMapMarker>
    <AIRMapMarker
      coordinate={
        Object {
          "latitude": undefined,
          "longitude": undefined,
        }
      }
      description="caption3"
      icon="file://assets/full/path/to/directory/name.type"
      onPress={[Function]}
      stopPropagation={false}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "position": "absolute",
          },
          undefined,
        ]
      }
      title="test 3"
    >
      <AIRMapCallout
        alphaHitTest={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "position": "absolute",
            },
            undefined,
          ]
        }
        tooltip={true}
      >
        <View>
          <View
            style={
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#fff",
                "borderColor": "#ccc",
                "borderRadius": 6,
                "borderWidth": 0.5,
                "flexDirection": "column",
                "padding": 15,
                "width": 150,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
            >
              3
              . 
              test 3
            </Text>
            <Text>
              caption3
            </Text>
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "borderTopColor": "#007a87",
                "borderWidth": 16,
                "marginTop": -0.5,
              }
            }
          />
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "borderTopColor": "#fff",
                "borderWidth": 16,
                "marginTop": -32,
              }
            }
          />
        </View>
      </AIRMapCallout>
    </AIRMapMarker>
    <AIRMapMarker
      coordinate={
        Object {
          "latitude": undefined,
          "longitude": undefined,
        }
      }
      description="caption4"
      icon="file://assets/full/path/to/directory/name.type"
      onPress={[Function]}
      stopPropagation={false}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "position": "absolute",
          },
          undefined,
        ]
      }
      title="test 4"
    >
      <AIRMapCallout
        alphaHitTest={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "position": "absolute",
            },
            undefined,
          ]
        }
        tooltip={true}
      >
        <View>
          <View
            style={
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#fff",
                "borderColor": "#ccc",
                "borderRadius": 6,
                "borderWidth": 0.5,
                "flexDirection": "column",
                "padding": 15,
                "width": 150,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
            >
              4
              . 
              test 4
            </Text>
            <Text>
              caption4
            </Text>
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "borderTopColor": "#007a87",
                "borderWidth": 16,
                "marginTop": -0.5,
              }
            }
          />
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "borderTopColor": "#fff",
                "borderWidth": 16,
                "marginTop": -32,
              }
            }
          />
        </View>
      </AIRMapCallout>
    </AIRMapMarker>
    <AIRMapMarker
      coordinate={
        Object {
          "latitude": undefined,
          "longitude": undefined,
        }
      }
      description="caption5"
      icon="file://assets/full/path/to/directory/name.type"
      onPress={[Function]}
      stopPropagation={false}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "position": "absolute",
          },
          undefined,
        ]
      }
      title="test 5"
    >
      <AIRMapCallout
        alphaHitTest={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "position": "absolute",
            },
            undefined,
          ]
        }
        tooltip={true}
      >
        <View>
          <View
            style={
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#fff",
                "borderColor": "#ccc",
                "borderRadius": 6,
                "borderWidth": 0.5,
                "flexDirection": "column",
                "padding": 15,
                "width": 150,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
            >
              5
              . 
              test 5
            </Text>
            <Text>
              caption5
            </Text>
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "borderTopColor": "#007a87",
                "borderWidth": 16,
                "marginTop": -0.5,
              }
            }
          />
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "borderTopColor": "#fff",
                "borderWidth": 16,
                "marginTop": -32,
              }
            }
          />
        </View>
      </AIRMapCallout>
    </AIRMapMarker>
    <AIRMapMarker
      coordinate={
        Object {
          "latitude": undefined,
          "longitude": undefined,
        }
      }
      description="caption6"
      icon="file://assets/full/path/to/directory/name.type"
      onPress={[Function]}
      stopPropagation={false}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "position": "absolute",
          },
          undefined,
        ]
      }
      title="test 6"
    >
      <AIRMapCallout
        alphaHitTest={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "position": "absolute",
            },
            undefined,
          ]
        }
        tooltip={true}
      >
        <View>
          <View
            style={
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#fff",
                "borderColor": "#ccc",
                "borderRadius": 6,
                "borderWidth": 0.5,
                "flexDirection": "column",
                "padding": 15,
                "width": 150,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
            >
              6
              . 
              test 6
            </Text>
            <Text>
              caption6
            </Text>
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "borderTopColor": "#007a87",
                "borderWidth": 16,
                "marginTop": -0.5,
              }
            }
          />
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "borderTopColor": "#fff",
                "borderWidth": 16,
                "marginTop": -32,
              }
            }
          />
        </View>
      </AIRMapCallout>
    </AIRMapMarker>
    <AIRMapMarker
      coordinate={
        Object {
          "latitude": undefined,
          "longitude": undefined,
        }
      }
      description="caption7"
      icon="file://assets/full/path/to/directory/name.type"
      onPress={[Function]}
      stopPropagation={false}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "position": "absolute",
          },
          undefined,
        ]
      }
      title="test 7"
    >
      <AIRMapCallout
        alphaHitTest={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "position": "absolute",
            },
            undefined,
          ]
        }
        tooltip={true}
      >
        <View>
          <View
            style={
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#fff",
                "borderColor": "#ccc",
                "borderRadius": 6,
                "borderWidth": 0.5,
                "flexDirection": "column",
                "padding": 15,
                "width": 150,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
            >
              7
              . 
              test 7
            </Text>
            <Text>
              caption7
            </Text>
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "borderTopColor": "#007a87",
                "borderWidth": 16,
                "marginTop": -0.5,
              }
            }
          />
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "borderTopColor": "#fff",
                "borderWidth": 16,
                "marginTop": -32,
              }
            }
          />
        </View>
      </AIRMapCallout>
    </AIRMapMarker>
  </AIRMap>
</View>
`;
